test/python/test_sampler.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+import pytest
+from qiskit import QuantumCircuit
+from qiskit.circuit.library import RealAmplitudes
+from qiskit.primitives import SamplerResult
+
+from mqt.ddsim.sampler import Sampler
+
+
+@pytest.fixture()
+def sampler() -> Sampler:
+    """The sampler fixture for the tests in this file."""
+    return Sampler()
+
+
+@pytest.fixture()
+def circuits() -> list[QuantumCircuit]:
+    """The circuit lists fixture for the tests in this file."""
+
+    bell_1 = QuantumCircuit(2)
+    bell_1.h(0)
+    bell_1.cx(0, 1)
+    bell_1.measure_all()
+
+    bell_2 = QuantumCircuit(2)
+    bell_2.h(0)
+    bell_2.cx(0, 1)
+    bell_2.x(1)
+    bell_2.measure_all()
+
+    param_qc_1 = RealAmplitudes(num_qubits=2, reps=2)
+    param_qc_1.measure_all()
+
+    param_qc_2 = RealAmplitudes(num_qubits=2, reps=3)
+    param_qc_2.measure_all()
+
+    return [bell_1, bell_2, param_qc_1, param_qc_2]
+
+
+def compare_probs(prob: list[dict] | dict, target: list[dict] | dict):
+    if not isinstance(prob, list):
+        prob = [prob]
+    if not isinstance(target, list):
+        target = [target]
+
+    assert len(prob) == len(target)
+
+    for p, targ in zip(prob, target):
+        for key, t_val in targ.items():
+            if key in p:
+                assert abs(p[key] - t_val) < 0.1
+            else:
+                assert abs(t_val) < 0.1
```

Comment:
Such kind of tolerance should be configurable IMHO. Could you please extract that into a function parameter with a reasonable default value (probably something smaller than `0.1`). You can just do a larger number of shots to reach that precision. These test circuits are so small that `1e6` shots would be really fast.